File: C++/fizzBuzz.cc
#include <iostream>
#include <string>
#include <vector>

class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ans;         //init Vector of string type
                                    // = dynamic arrays and allow to add and remove anytime
        for (int i = 1; i <= n; i++){
            if (i % 3 == 0 && i % 5 == 0){
                ans.push_back("FizzBuzz");      //string can add at anytime using push_back
            } else if (i % 3 == 0){
                ans.push_back("Fizz");
            } else if (i % 5 == 0){
                ans.push_back("Buzz");
            } else {
                ans.push_back(to_string(i));
            }
        }
        return ans;
    }
};

File: Python/algorithms/fizzBuzz.py
#elegant way: String concatenation
# allow to solve extended porblem
# ie FizzBuzzJazz

from typing import List


class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        #return list
        ans = []

        for num in range (1, n+1):
            div_by_3 = (num % 3 == 0)
            div_by_5 = (num % 5 == 0)

            curr = ""

            if div_by_3:
                # Divides by 3
                curr += "Fizz"
            if div_by_5:
                # Divides by 5
                curr += "Buzz"
            if not curr: 
                #if current string empty (ie not divides by any of above)
                curr = str(num)
            # appent the current string to the answer list
            ans.append(curr)
        
        return ans

